src/generate-services.ts

```diff
@@ -57,11 +58,18 @@ export function generateService(
       params.push(code`ctx: Context`);
     }
 
-    // the grpc-web clients auto-`fromPartial` the input before handing off to grpc-web's
-    // serde runtime, so it's okay to accept partial results from the client
-    const partialInput = options.outputClientImpl === 'grpc-web';
-    const inputType = requestType(ctx, methodDesc, partialInput);
-    params.push(code`request: ${inputType}`);
+    // grpc-web client stream or bidirectional stream not request params when use Promise.
+    if (!options.returnObservable && options.outputClientImpl === 'grpc-web' && methodDesc.clientStreaming) {
```

Comment:
Just invert the if statement here.

if (options.returnObservable || options.outputClientImpl !== "grpc-web" || !methodDesc.clientStreaming)

Reply:
"grpc-web client stream or bidirectional stream not request params when use Promise."

I can't understand this sentence, could you make it more clear?

Reply:
When using the `Promise` return type, the input parameter does not require the `request` parameter.

See https://github.com/stephenh/ts-proto/pull/631/files#diff-cf75bd2b26f57126e3c440377e55dfa55c406359a04d2029355db727cac33c3eR44

In this implementation, the `request` parameter is not required. Use `callback` to get the relevant data stream, use the `write` method to send data streams.